### PR TITLE
Changed vw_qgep_cover_INSERT and vw_qgep_cover_UPDATE by adding NEW.i…

### PR DESCRIPTION
**Pull request by bogdanvaduva**
Hi m_kuhn you suggested me 2 weeks ago a pull request for setting a default value for identifier.
Sorry for getting back to you so late. Better to be late than ever :D.
Changed vw_qgep_cover_INSERT and vw_qgep_cover_UPDATE by adding NEW.identifier value if null. Also added tests.
I hope it's what you had in mind.
Regards

### CODE REVIEW

File: test/test_triggers.py [hunk 1]
```diff
@@ -51,5 +51,18 @@ def test_last_modified(self):
...
+                'identifier': '',
+                'level': decimal.Decimal('50.000')
+        }
+
+        obj_id = self.insert_check('vw_cover', row)
+
+        row = self.select('od_structure_part', obj_id)
+
+        identifier = row['identifier']
+        assert identifier is not None, "Identifier not set on insert"
```
Comment (m-kuhn) [inline test/test_triggers.py]: Shouldn't it also check that it's not empty? At least there's an empty string specified above which should be replaced by the obj_id and here the check is for None.

Does

    assert identifier, "Identifier not set on insert"

work? In python a string converted to a boolean is False if it's empty or None IIRC.

Comment (m-kuhn) [inline test/test_triggers.py]: Or even better `assert identifier, "Identifier not set on insert: '{}'".format(repr(identifier))`

Comment (m-kuhn) [inline test/test_triggers.py]: Can you do this small update? If the test still passes after that we can merge this important update.

**Comment by bogdanvaduva** [comment]
It doesn't work with assert identifier, .... It has to be assert is not None. At least it is what travis gives me.
Regards

**Comment by m-kuhn** [comment]
Doesn't that mean that the test fails for a valid reason?

On [line 56](https://github.com/QGEP/datamodel/pull/3/files#diff-98e9907bc70fc483e2504f48edf95fd6R56) a new entry is inserted with an empty `''` identifier, on [line 65](https://github.com/QGEP/datamodel/pull/3/files#diff-98e9907bc70fc483e2504f48edf95fd6R65) the check asserts that the new identifier is not `NULL`. So the test succeeds if the identifier is properly updated (=set to the obj_id) but also if nothing at all happens (the identifier is still empty).

My assumption is at the moment is the second case: A new record with an empty identifier is inserted, the trigger does not work for some reason, the check for None reports positive. But it's not what we really want to happen.

**Comment by bogdanvaduva** [comment]
Yes, you were right about your assumption. Here was the problem: I didn't changed vw_cover_insert trigger. Now I did changed vw_cover_insert. I also changed test_trigger.py. 

You have to bear with me, because I am not a python programmer so everything is new.

**Comment by m-kuhn** [comment]
Manually merged in ede1f3ec484d9e65a062ce0b40c617ecd1edf79a

I realized that changes to vw_cover are useless since the file is generated and therefore changes will not be long-lasting. So the only merged change concerns vw_qgep_cover which is manually managed.

If changes to vw_cover should be required as well they'll need to be implemented differently. But if that's required there's probably a lot more to it (i.e. more generic). I hope the current fix is all that's needed.

Thanks a lot @bogdanvaduva!

**Comment by sjib** [comment]
Tell me what I have to change in the Metadatabase export for vw_cover – then I will implement it.

**Comment by m-kuhn** [comment]
I wonder if we should really introduce this kind of magic. For me it's fine at the moment.

The question would be: When inserting a second cover on an existing wastewater structure, should the identifier automatically be synchronized with its obj_id when left empty?
And if this question is answered with yes, then it should probably apply to all objects with identifiers. I think it's an overkill. Therefore if it's for me, leave it as is.

**Comment by sjib** [comment]
I think also this is P2 or P3 – nice to have but not worth a lot of time and money right now.

**Pull request closed without merging**